src/user-management/user-management.spec.ts

```diff
@@ -352,6 +354,147 @@ describe('UserManagement', () => {
     });
   });
 
+  describe('authenticateWithSessionCookie', () => {
+    beforeEach(() => {
+      // Mock createRemoteJWKSet
+      jest
+        .spyOn(jose, 'createRemoteJWKSet')
+        .mockImplementation(
+          (_url: URL, _options?: jose.RemoteJWKSetOptions) => {
+            // This function simulates the token verification process
+            const verifyFunction = (
+              _protectedHeader: jose.JWSHeaderParameters,
+              _token: jose.FlattenedJWSInput,
+            ): Promise<jose.KeyLike> => {
+              return Promise.resolve({
+                type: 'public',
+              });
+            };
+
+            // Return an object that includes the verify function and the additional expected properties
+            return {
+              __call__: verifyFunction,
+              coolingDown: false,
+              fresh: false,
+              reloading: false,
+              reload: jest.fn().mockResolvedValue(undefined),
+              jwks: () => undefined,
+            } as unknown as ReturnType<typeof jose.createRemoteJWKSet>;
+          },
+        );
```

Comment:
In general testing with JWKs is a bit cumbersome, and I feel like where I've ended up here the tests are a bit weak.